PRCS/prc-4.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Paima Orderbook DEX
+description: Interface for facilitating trading of a game asset (living on a game chain) on different base chain.
+author: sebastiengllmt, matejos (@matejos)
+status: Draft
+created: 2024-03-13
+---
+
+## Abstract
+
+Allowing tradability of game assets directly on popular networks helps achieve a lot more composability and liquidity than would otherwise be possible. This standard helps define how to define an orderbook-like DEX smart contract for trading game assets on different chains without introducing centralization, lowered security or wait times for finality.
+
+## Motivation
+
+Many games, due to being data and computation heavy applications, run on sidechains, L2s and appchains as opposed to popular L1 blockchains. This is problematic because liquidity for trading assets live primarily on the L1s (different environments). A common solution to this problem is building a bridge, but bridges have a bad reputation, often require a long delay (especially for optimistic bridges which often require 1 week), and bridging also makes upgrading the game harder as any update to the game state may now also require you to update the data associated with all the bridged state.
+
+Instead of bridging, this standard allows trading the game assets on base chain via creating and filling sell orders in an orderbook DEX smart contract.
+
+## Specification
+
+Every PRC-4 compliant contract must implement the `IOrderbookDex` interface and events:
+
+```solidity
+/// @notice Facilitates base-chain trading of an asset that is living on a different app-chain.
+/// @dev The contract should never hold any ETH itself.
+interface IOrderbookDex is IERC165 {
```

Comment:
We'll need a baseURI specified for this similar to the baseURI format we defined in PRC3 since Tarochi/Paima will have to implement that (as well as a function in the interface that is equivalent to tokenURI)